core/input/input.h

```diff
@@ -53,6 +81,28 @@ class Input : public Object {
 		MOUSE_MODE_CONFINED_HIDDEN,
 	};
 
+	enum BufferingMode {
+		BUFFERING_MODE_NONE,
+		BUFFERING_MODE_FRAME,
+		BUFFERING_MODE_AGILE,
```

Comment:
A description here would help. At least I'd be grateful of one, for reviewing purposes.